File: examples/code/integrations/notion/custom_auth.js
import { Arcade } from '@arcadeai/arcadejs';

const client = new Arcade();

const userId = '{arcade_user_id}';

// Start the authorization process
const authResponse = await client.auth.start(userId, 'notion');

if (authResponse.status !== 'completed') {
  console.log('Please complete the authorization challenge in your browser:');
  console.log(authResponse.url);
}

// Wait for the authorization to complete
authResponse = await client.auth.waitForCompletion(authResponse);

const token = authResponse.context.token;
// Do something interesting with the token...

File: examples/code/integrations/miro/custom_auth.py
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

user_id = "{arcade_user_id}"

# Start the authorization process
auth_response = client.auth.start(
    user_id=user_id,
    provider="arcade-miro",
    scopes=["boards:read", "boards:write"],
)

if auth_response.status != "completed":
    print("Please complete the authorization challenge in your browser:")
    print(auth_response.url)

# Wait for the authorization to complete
auth_response = client.auth.wait_for_completion(auth_response)

if not auth_response.context.token:
    raise ValueError("No token found in auth response")

token = auth_response.context.token
# TODO: Do something interesting with the token...

